nalgebra-sparse/src/csr.rs

```diff
@@ -719,6 +721,14 @@ macro_rules! impl_csr_row_common_methods {
             pub fn get_entry(&self, global_col_index: usize) -> Option<SparseEntry<'_, T>> {
                 self.lane.get_entry(global_col_index)
             }
+
+            /// Iterator over the column indices and elements of a row of a CSR matrix.
+            /// Equivalent to `row.col_indices().iter().zip(row.values().iter())`.
+            #[inline]
+            #[must_use]
+            pub fn iter(&self) -> CsInnerIter<'_, T> {
```

Comment:
I'm wondering if perhaps it would be better to have a more descriptive name, to indicate what precisely is being iterated over. Otherwise I feel as if code using this iterator might be hard to read, since the reader might not have any clue what *precisely* is being iterated over? What about  `col_and_value_iter`?

On the other hand, I guess it also comes down to how the user calls the iterator. For example,
```rust
for (col_idx, value) in row.iter() {}
```
is pretty clear. In fact, we could maybe even implement `IntoIterator` :thinking:
```rust
for (col_idx, value) in row {}
```
but we can leave this for the future though!

Do you have any thoughts on the naming?

Reply:
I'm also thinking maybe we should do like e.g. `CsrRow` and `CscCol` and wrap `CsInnerIter` in CSR/CSC-specific types, so that we can provide more specific documentation.

One thing I've tried hard to do in `nalgebra-sparse` is to make it easy to understand how things work. So if you want to use a CSR matrix you shouldn't need to know about CSC matrices. In particular, the generalization of rows/cols to lanes is something I've mostly tried to "hide" in the API of CSR/CSC, because this generalization makes it harder to understand. This means of course that we pay with this with some slight duplication in the APIs. What's your opinion on this?

Reply:
I like your idea of hiding the lane iterator behind both a specific row and column version. I agree that makes things more clear and has little cost. I also like the idea of implementing 'IntoIterator', I don't mind changing the name if you think the clarity is needed here as well.